datafusion/sql/src/select.rs

```diff
@@ -431,6 +431,51 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         group_by_exprs: Vec<Expr>,
         aggr_exprs: Vec<Expr>,
     ) -> Result<(LogicalPlan, Vec<Expr>, Option<Expr>)> {
+        let schema = input.schema();
+        let id_key_groups = schema.identifier_key_groups();
```

Comment:
here is another good candidate to put into a method on `IdentifierKeyGroups`  with a named function which I think would keep the sql planner easier to read and understand what was happening